src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -5,20 +5,116 @@
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int buckets;
+    private Node[] table;
+
+    public MyHashMap() {
+        this.table = new Node[INITIAL_CAPACITY];
+        this.buckets = 0;
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        addNode(key, value, table);;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
+            return null;
+        }
+        Node searchNode = table[indexFor(key, table.length)];
```

Comment:
let's make this first and then make check whether searchNode == null. This way we can get rid of the checking above.

Reply:
Done